src/EditorFeatures/CSharp/MisplacedUsings/MisplacedUsingsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+                && !ShouldSurpressDiagnostic(compilationUnit))
+            {
+                ReportDiagnostics(context, _insideDescriptor, compilationUnit.Usings, option);
+            }
+            else if (statement is NamespaceDeclarationSyntax namespaceDeclaration
+                && option.Value == UsingPlacementPreference.OutsideNamespace)
+            {
+                ReportDiagnostics(context, _outsideDescriptor, namespaceDeclaration.Usings, option);
+            }
+        }
+
+        private CodeStyleOption<UsingPlacementPreference> GetPreferredPlacementOption(SyntaxNodeAnalysisContext context)
+        {
+            var statement = context.Node;
+            var cancellationToken = context.CancellationToken;
+
+            // While running in VisualStudio this should always return an OptionSet.
+            var optionSet = context.Options.GetDocumentOptionSetAsync(statement.SyntaxTree, cancellationToken).GetAwaiter().GetResult();
+            if (optionSet != null)
+            {
+                return optionSet.GetOption(CSharpCodeStyleOptions.PreferredUsingPlacement);
+            }
+
+            // This code path is for tests only. It relies on the test creating a .editorconfig in the current working directory.
+            var filePath = context.SemanticModel.SyntaxTree.FilePath;
+            var fullPath = string.IsNullOrEmpty(Path.GetDirectoryName(filePath))
+                ? Path.Combine(Environment.CurrentDirectory, filePath)
+                : filePath;
+
+            // Find the correct key name from the .editorconfig storage location
+            var storageLocation = CSharpCodeStyleOptions.PreferredUsingPlacement.StorageLocations
+                .OfType<EditorConfigStorageLocation<CodeStyleOption<UsingPlacementPreference>>>()
+                .FirstOrDefault();
+
+            var conventionContext = _conventionsManager.GetConventionContextAsync(fullPath, cancellationToken).GetAwaiter().GetResult();
+            if (conventionContext.CurrentConventions.TryGetConventionValue(storageLocation.KeyName, out string usingPlacementPreference))
+            {
+                return CSharpCodeStyleOptions.ParseUsingPlacementPreference(usingPlacementPreference, s_noPreferenceOption);
+            }
+
+            return s_noPreferenceOption;
+        }
+
+        private bool ShouldSurpressDiagnostic(CompilationUnitSyntax compilationUnit)
+        {
+            // Surpress if file contains a type declaration in the global namespace
+            // or if an attribute is used in the global namespace.
+            return compilationUnit.ChildNodes().Any(node =>
+                node.IsKind(SyntaxKind.ClassDeclaration)
+                || node.IsKind(SyntaxKind.InterfaceDeclaration)
+                || node.IsKind(SyntaxKind.EnumDeclaration)
+                || node.IsKind(SyntaxKind.StructDeclaration)
+                || node.IsKind(SyntaxKind.DelegateDeclaration)
+                || node.IsKind(SyntaxKind.AttributeList));
```

Comment:
top level might also include things like members.  instead of a blacklist, make a whitelist.